bottlerocket-settings-models/settings-extensions/bootstrap-commands/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+//! Settings related to bootstrap commands.
+use bottlerocket_model_derive::model;
+use bottlerocket_modeled_types::{ApiCommand, BootstrapCommandMode, Identifier};
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::HashMap, convert::Infallible};
+
+/// BootstrapCommandsSettingsV1 represents how a BootstrapCommand lives as a Bottlerocket Setting.
+/// It can be set by either user-data or through Bottlerocket API. Lets take an example
+/// [settings.bootstrap-commands."001-test-bootstrap-commands"]
+/// commands = [["apiclient", "set", "motd=helloworld1"]]
+/// mode = "always"
+/// essential = true
+/// Here, the name (Identifier) is "001-test-bootstrap-commands" and commands, mode and essential
+/// form the BootstrapCommand object.
+/// The Bootstrap Commands are executed in lexicographical order of their names (Identifier).
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct BootstrapCommandsSettingsV1 {
+    pub bootstrap_commands: HashMap<Identifier, BootstrapCommand>,
+}
+
+// Custom serializer/deserializer added to maintain backwards
+// compatibility with models created prior to settings extensions.
+impl Serialize for BootstrapCommandsSettingsV1 {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.bootstrap_commands.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for BootstrapCommandsSettingsV1 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let bootstrap_commands = HashMap::deserialize(deserializer)?;
+        Ok(Self { bootstrap_commands })
+    }
+}
+
+/// BootstrapCommand represents the set of settings within a Bootstrap Command
+/// The commands within a BootstrapCommand are run serially one after the other.
```

Comment:
This property - serial execution - is not something that this code enforces or cares about, which makes the comment distracting and unhelpful.

Reply:
I advocated to add these because I was missing this piece of the mental puzzle, but you're right: it's not warranted here since the model does not dictate it be used this way.